src/shims/unix/socket.rs

```diff
@@ -39,15 +54,71 @@ impl FileDescription for SocketPair {
         "socketpair"
     }
 
+    fn return_epoll_events<'tcx>(&mut self) -> InterpResult<'tcx, &mut Vec<Weak<EpollEvent>>> {
+        Ok(&mut self.epoll_events)
+    }
+
+    fn check_readiness<'tcx>(&self, ecx: &mut MiriInterpCx<'tcx>) -> InterpResult<'tcx, Vec<u32>> {
+        // We only check the status of EPOLLIN, EPOLLOUT and EPOLLRDHUP flag. If other event flags
+        // need to be supported in the future, the check should be added here.
+        let epollin = ecx.eval_libc_u32("EPOLLIN");
+        let epollout = ecx.eval_libc_u32("EPOLLOUT");
+        let epollrdhup = ecx.eval_libc_u32("EPOLLRDHUP");
+        let mut ready_flags = Vec::new();
+        let readbuf = self.readbuf.borrow();
+        // Check if the writeend has closed.
```

Comment:
```suggestion
        // Check if the write end has been closed.
```

Reply:
This check for epollrdhup is replaced by the check below that based on the closure of peer socketpair file description.